### PR TITLE
feat: add BatchNormalization layer full-stack support

### PR DESCRIPTION
**Pull request by paramananda-15**
## Description

Adds support for the BatchNormalization layer in both backend and frontend.

This enables users to include BatchNorm layers in the visual model builder,
which are mapped to tf.keras.layers.BatchNormalization during model generation.

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

## How Has This Been Tested?

- Backend tested using:
  uv run pytest

- Frontend validated using:
  npx eslint --fix . --ext js,jsx
  npx prettier --write "src/**/*.{js,jsx,json,css}"

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review
- [x] My changes generate no new warnings
- [x] Tests pass locally

### CODE REVIEW

**Comment by ivantha** [comment]
Closing as stale during repository cleanup. Thanks for the contribution. Feel free to reopen against the latest main if you'd like this revisited.
